### ISSUE TITLE
Dumping pickled data

### ISSUE THREAD

**Issue by shon** [iamteem/redisco#17]
```
class Request(models.Model):
    data = models.Attribute()

req = Request()
req.data = cPickle.dumps(data, -1)
req.save()
```

Not sure why unicode conversion is done while saving and retrieving. 
It work perfectly for me if I by patch the code where patch catches the exception and returns the value without unicode cast.

**Comment by iamteem** [comment]
Hi shon, can you point me to your patch? I wanna see. Thanks.

**Comment by shon** [comment]
It's nothing much. Let me rephrase and call it a hack not patch ;). Here it goes

```
--- attributes.py   2011-04-07 21:42:29.922536002 +0530
+++ /home/shon/work/cowspa/lib/python2.6/site-packages/redisco/models/attributes.py 2011-04-07 16:34:05.513716806 +0530
@@ -65,14 +65,20 @@
     def typecast_for_read(self, value):
         """Typecasts the value for reading from Redis."""
         # The redis client encodes all unicode data to utf-8 by default.
-        return value.decode('utf-8')
+        try:
+            return value.decode('utf-8')
+        except:
+            return value

     def typecast_for_storage(self, value):
         """Typecasts the value for storing to Redis."""
         try:
             return unicode(value)
         except UnicodeError:
-            return value.decode('utf-8')
+            try:
+                return value.decode('utf-8')
+            except:
+                return value

     def value_type(self):
         return unicode
```
